src/uu/env/src/env.rs

```diff
@@ -323,6 +323,22 @@ impl EnvAppData {
     fn run_env(&mut self, original_args: impl uucore::Args) -> UResult<()> {
         let (original_args, matches) = self.parse_arguments(original_args)?;
 
+        // clap automatically removes '=' from the beginning of the argument, so we need to check for it
+        for arg in &original_args {
+            let arg = arg.to_string_lossy();
+            if arg.starts_with("-u=") || arg.starts_with("--unset=") {
+                let invalid_arg = if arg.starts_with("-u=") {
+                    &arg[2..]
+                } else {
+                    &arg[6..]
+                };
+                return Err(UUsageError::new(
+                    125,
+                    format!("cannot unset '{}': Invalid argument", invalid_arg),
+                ));
+            }
+        }
```

Comment:
it can be simplified by something like:

```suggestion
for arg in &original_args {
    let arg = arg.to_string_lossy();
    let prefix = if arg.starts_with("-u=") { "-u=" } else { "--unset=" };
    if arg.starts_with(prefix) {
        let invalid_arg = &arg[prefix.len()..];
        return Err(UUsageError::new(
            125,
            format!("cannot unset '{}': Invalid argument", invalid_arg),
        ));
    }
}

```

Reply:
Thanks that is much cleaner. But just letting you know that
`let invalid_arg = &arg[prefix.len()..];`
needs to be
`let invalid_arg = &arg[(prefix.len())-1..];`